pyfr/backends/base/types.py

```diff
@@ -102,12 +104,15 @@ def _pack(self, ary):
 
     def _unpack(self, ary):
         # Unpack from [blocked] AoSoA to SoA
-        ary = ary.swapaxes(0, 1)
+        ary = ary.reshape(self.datashape)
 
         if len(self.ioshape) > 2:
-            ary = ary.reshape(self.datashape)
             ary = ary.swapaxes(-2, -3)
```

Comment:
Cab we not do a transpose to undo the blocking + AoSoA in a single call?  Then a single reshape to get it close to the I/O shape but with padding?